pubtools/pulplib/_impl/model/unit/modulemd.py

```diff
@@ -128,7 +133,13 @@ class ModulemdUnit(Unit):
 
     """
 
-    profiles = pulp_attrib(type=dict, pulp_field="profiles", default=None)
+    profiles = pulp_attrib(
+        type=dict,
+        pulp_field="profiles",
+        default=None,
+        validator=optional_dict_or_frozendict,
```

Comment:
This confused me because tests complained that frozendict wasn't a dict, despite the class definition clearly inheriting from dict. 

Digging deeper now, I found that frozendict has two definitions. The default definition imports a cpython class that inherits from object, so it wasn't actually a dict. I think this one is preferred because it's faster. The dict version is treated as a fallback if there's something wrong with importing cpython, but can just be imported directly. I've imported it now and changed the validator function